persistent/Database/Persist/Class/PersistUnique.hs

```diff
@@ -571,6 +572,49 @@ checkUniqueKeys (x:xs) = do
         Nothing -> checkUniqueKeys xs
         Just _ -> return (Just x)
 
+-- | Check whether there are any conflicts for unique keys with this entity and
+-- existing entities in the database.
+--
+-- Returns 'Nothing' if the entity would stay unique, and could thus safely be updated.
+-- on a conflict returns the conflicting key
+--
+-- This is similar to 'checkUnique', except it's useful for updating - when the
+-- particular entity already exists, it would normally conflict with itself.
+-- This variant ignores those conflicts
+--
+-- === __Example usage__
+--
+-- We use <#schema-persist-unique-1 schema-1> and <#dataset-persist-unique-1 dataset-1> here.
+--
+-- This would be 'Nothing':
+--
+-- > mAlanConst <- checkUnique $ User "Alan" 70
+--
+-- While this would be 'Just' because SPJ already exists:
+--
+-- > mSpjConst <- checkUnique $ User "SPJ" 60
```

Comment:
Awesome docs, thanks :smile:

Reply:
I can't take the credit - these are largely copied from `checkUnique`, plus a paragraph on the difference in behaviour.